### PR TITLE
FEATURE: MSF-24679 LCM Rollout brick allows hard sync

### PR DESCRIPTION
**Pull request by hung-nguyen-hoang**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1308/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1178/ : FAILURE in 9m 50s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1310/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1180/ : FAILURE in 8m 51s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1311/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1181/ : FAILURE in 8m 35s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1312/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1182/ : FAILURE in 8m 40s

**Comment by hung-nguyen-hoang** [comment]
ok to test

**Comment by hung-nguyen-hoang** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1314/ : SUCCESS in 11s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1184/ : FAILURE in 9m 12s

**Comment by hung-nguyen-hoang** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1315/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1185/ : FAILURE in 9m 10s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1316/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1186/ : FAILURE in 9m 02s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1318/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1188/ : FAILURE in 9m 09s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1320/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1190/ : SUCCESS in 9m 55s

**Comment by tqtu** [comment]
ok to test

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1381/ : SUCCESS in 5s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1249/ : SUCCESS in 10m 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1402/ : SUCCESS in 5s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1267/ : SUCCESS in 11m 54s

**Comment by tqtu** [comment]
Verified on Staging2

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1418/ : SUCCESS in 5s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1283/ : SUCCESS in 10m 33s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1423/ : SUCCESS in 4s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1288/ : SUCCESS in 10m 38s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1426/ : SUCCESS in 4s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1291/ : SUCCESS in 12m 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1427/ : SUCCESS in 4s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1292/ : SUCCESS in 10m 27s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1428/ : SUCCESS in 4s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1293/ : SUCCESS in 10m 22s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1429/ : SUCCESS in 4s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1294/ : SUCCESS in 10m 15s

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 1]
```diff
@@ -51,8 +51,6 @@ def collect_synced_status(params)
           params.include?(COLLECT_SYNCED_STATUS) && to_bool(COLLECT_SYNCED_STATUS, params[COLLECT_SYNCED_STATUS])
         end
 
-        private
```
Comment (danh-ung) [inline lib/gooddata/lcm/helpers/check_helper.rb:54]: irrelevant change?

File: lib/gooddata/models/project_creator.rb [hunk 2]
```diff
@@ -204,6 +210,52 @@ def pick_correct_chunks(chunks, opts = {})
           end
         end
 
+        def pick_correct_chunks_hard_sync(chunks, opts = {})
+          GoodData.logger.debug("update_preference") { opts[:update_preference].pretty_inspect }
```
Comment (danh-ung) [inline lib/gooddata/models/project_creator.rb:214]: we should remove GoodData.logger.debug before merging.

Comment (hung-nguyen-hoang) [reply lib/gooddata/models/project_creator.rb]: Done remove it. We also don't need log them for each project.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1441/ : SUCCESS in 5s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1306/ : SUCCESS in 11m 47s

File: lib/gooddata/lcm/helpers/check_helper.rb (hunk 1 above)
Comment (hung-nguyen-hoang) [reply lib/gooddata/lcm/helpers/check_helper.rb:54]: No, It relates to the change. Remove it to public method below it.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1474/ : SUCCESS in 4s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1339/ : SUCCESS in 10m 24s

**Comment by danh-ung** [comment]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ruby-gate-PR1902

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-promote-pipeline/187/ : SUCCESS in 14m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-gate/124/ : SUCCESS in 2m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-fossa-scanning-zuul/16/ : FAILURE in 6m 28s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/666/ : SUCCESS in 14s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-gitops-deploy-pipeline/89/ : SUCCESS in 59s

**Pull request merged**